skimage/transform/tests/test_thin_plate_splines_transform.py

```diff
@@ -45,63 +35,103 @@ def test_tps_transform_ensure_2d():
     assert_array_equal(_ensure_2d(array_1d), expected)
 
     empty_array = np.array([])
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match="Array of points can not be empty."):
         _ensure_2d(empty_array)
 
     scalar = 5
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match="Array must be be 2D."):
         _ensure_2d(scalar)
 
     array_3d = np.array([[[0, 5], [10, 15]], [[20, 25], [30, 35]]])
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match="Array must be be 2D."):
         _ensure_2d(array_3d)
 
     control_pts_less_than_3 = np.array([[0, 0], [0, 0]])
-    with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match="Array points less than 3 is undefined."):
         _ensure_2d(control_pts_less_than_3)
 
+def test_tps_transform_init():
+    tform = TpsTransform()
+
+    # Test that _estimated is initialized to False
+    assert tform._estimated is False
+    assert tform.parameters is None
+    assert tform.control_points is None
 
 def test_tps_transform_estimation():
-    tform = TPSTransform()
-    assert tform.estimate(DST, SRC)
+    tform = TpsTransform()
+
+    # Ensure that the initial state is as expected
+    assert tform._estimated is False
+    assert tform.parameters is None
+    assert tform.control_points is None
+
+    # Perform estimation
+    tcoeffs = tform.estimate(SRC, DST)
+
+    # Check if the estimation was successful
+    assert tcoeffs
     assert tform._estimated is True
     assert len(tform.control_points) > 0
 
     assert len(tform.parameters) > 0
-    assert tform.parameters.shape[0] ==  SRC.shape[0] + 3
+    assert tform.parameters.shape[0] == SRC.shape[0] + 3
     np.testing.assert_array_equal(tform.control_points, SRC)
 
-def test_tps_transform_init():
-    tps = TPSTransform()
-
-    # Test that _estimated is initialized to False
-    assert tps._estimated is False
+def test_tps_transform_estimation_failure():
+    # Test the estimate method when the estimation fails
+    tform = TpsTransform()
+    src = np.array([[0, 0], [0, 5], [5, 5], [5, 0]])
+    dst = np.array([[5, 0], [0, 0], [0, 5]])
 
-    # Test that parameters is an empty array of dtype float32
-    assert isinstance(tps.parameters, np.ndarray)
-    assert tps.parameters.dtype == np.float32
-    assert tps.parameters.size == 0
+    # Ensure that the initial state is as expected
+    assert tform._estimated is False
+    assert tform.parameters is None
+    assert tform.control_points is None
 
-    # Test that control_points is an empty array of dtype float32
-    assert isinstance(tps.control_points, np.ndarray)
-    assert tps.control_points.dtype == np.float32
-    assert tps.control_points.size == 0
+    # Perform the estimation, which should fail due to the mismatched number of points
+    with pytest.raises(ValueError):
```

Comment:
```suggestion
    with pytest.raises(ValueError, match=".*shape must be identical"):
```
If it should fail due to X then we should test that.